types/k6/experimental/websockets.d.ts

```diff
@@ -52,16 +60,91 @@ export class WebSocket {
     /**
      * Bind event names to event handlers to be executed when their
      * respective event is received by the server.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener/
      *
      * @param event - the event to listen for
      * @param listener - the callback to invoke when the event is emitted
      */
-    addEventListener(event: EventName, listener: (message: MessageEvent) => void): void;
+    addEventListener(event: EventName, listener: (event: MessageEvent | ErrorEvent) => void): void;
 
     /**
      * Closes the WebSocket connection or connection attempt, if any.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-close/
+     *
+     * @param code - An integer WebSocket connection close code value indicating a reason for closure.
+     * @param reason - A human-readable string WebSocket connection close reason. No longer than 123 bytes of UTF-8 text.
+     */
+    close(code?: number, reason?: string): void;
+
+    /**
+     * Sends a ping message over the WebSocket connection.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-ping/
+     */
+    ping(): void;
+
+    /**
+     * Sets an event handler which is invoked when a message event happens.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-onmessage/
+     *
+     * @param event - the message event
+     */
+    onmessage: (event?: MessageEvent) => void;
+
+    /**
+     * Sets an event handler which is invoked when the WebSocket connection's opens.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-onopen/
+     */
+    onopen: () => void;
+
+    /**
+     * Sets an event handler which is invoked when the WebSocket connection's closes.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-onclose/
+     */
+    onclose: () => void;
+
+    /**
+     * Sets an event handler which is invoked when errors occur.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-onerror/
+     *
+     * @param event - the error event
+     */
+    onerror: (event?: ErrorEvent) => void;
+
+    /**
+     * Sets an event handler which is invoked when a ping message is received.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-onping/
+     */
+    onping: () => void;
+
+    /**
+     * Sets an event handler which is invoked when a pong message is received.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/websocket-onpong/
+     */
+    onpong: () => void;
+}
+
+/**
+ * k6 specific WebSocket parameters.
+ * https://k6.io/docs/javascript-api/k6-experimental/websockets/params/
+ */
+export interface Params {
+    /** Request headers. */
+    headers?: { [name: string]: string };
```

Comment:
I believe you should be able to replace this with the[ Record syntax](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) here: `headers?: Record<string, string>`